### DISCUSSION TITLE
Geodesic distance between two barycenters

### DISCUSSION THREAD

**Discussion by royvelich** [libigl/libigl#2041] [Q&A]
Hello,
I wonder if there is a way in LIBIGL to calculate the geodesic distance between two arbitrary points on a triangular mesh (given by their barycentric coordinates). From what I understand, the function exact_geodesic calculates the distance between two mesh vertices and/or face centers. I also understood that there might be a work-around to it, by adding the source and destination vertices to the mesh. However, I am not sure how can I do it, since it might require to perform a subdivision of the triangles to which the two vertices belong. Is there some kind of a high-level method in LIBIGL that can add those two vertices to the mesh for me?

Thanks.

**Comment by alecjacobson** [comment]
That's right. I think the simplest way would be temporarily insert the vertices into the mesh.
